models/Course.js

```diff
@@ -17,9 +21,14 @@ const Course = mongoose.model('Course', {
       message: 'Description must be valid length'
     }
   },
-  tags: [String],
+  tags: [TagSchema],
```

Comment:
I wonder if you think a simple String is enough to model our tag.
Just worried it adds extra complexity which you want to avoid.